server/block/fire.go

```diff
@@ -41,6 +42,22 @@ func neighboursFlammable(pos cube.Pos, w *world.World) bool {
 	return false
 }
 
+// Splash checks to see if the fire was splashed by a bottle and then extinguishes itself and its neighboring fires.
+func (f Fire) Splash(pos cube.Pos, p *entity.SplashPotion) {
+	if p.Type() != potion.Water() {
+		return
+	}
+	w := p.World()
+	w.SetBlock(pos, nil, nil)
+	for _, face := range cube.HorizontalFaces() {
+		h := pos.Side(face)
+		fire := Fire{}
+		if w.Block(h) == fire {
```

Comment:
Would use a type assertion here and also inline the `h` variable.

Reply:
The variable h is used twice though, its used to set the block.

Reply:
used type assertion, idk about inlining h though, can you give a suggestion? (I kinda suck at go rn, very rusty)

Reply:
> used type assertion, idk about inlining h though, can you give a suggestion? (I kinda suck at go rn, very rusty)

```
if h := pos.Side(face); w.Block(h) == fire {}
```

Reply:
Never mind, didn't realize h was used twice, I didn't mean it like that either @RoyalMCPE